src/test/integration/ConnectionsPerformanceTest.cpp

```diff
@@ -199,9 +198,13 @@ namespace nupic
       {
         segment = connections.getSegment(winnerCell, 0);
 
-        for (Synapse synapse : connections.synapsesForSegment(segment))
+        const vector<Synapse>& synapses =
+          connections.synapsesForSegment(segment);
+
+        for (SynapseIdx i = 0; i < (SynapseIdx)synapses.size();)
         {
-          synapseData = connections.dataForSynapse(synapse);
+          Synapse synapse = synapses[i];
```

Comment:
Agreed, const. (I'm intentionally keeping `Segment`s and `Synapse`s small and using them by value. They're just 32 bits, so a reference would actually be larger on a 64-bit machine.)